src/modules/tensor/hip/kernel/hue.cpp

```diff
@@ -0,0 +1,299 @@
+{
+    // Convert RGB to HSV
+    float hue, sat, val;
+    RGB_to_HSV_hip(pixelR, pixelG, pixelB, hue, sat, val);
+
+    // Apply hue adjustment
+    hue += *hueParam;
+        
+    // Normalize hue to 0-6 range
+    if (hue < 0) hue += 6.0f;
+    else if (hue >= 6.0f) hue -= 6.0f;
+
+    // Convert HSV to RGB 
+    HSV_to_RGB_hip(hue, sat, val, pixelR, pixelG, pixelB); 
+}
+
+__device__ void hue_8RGB_hip_compute(d_float24 *pix_f24, float *hueParam)
+{
+    hue_1RGB_hip_compute(&(pix_f24->f1[ 0]), &(pix_f24->f1[ 8]), &(pix_f24->f1[16]), hueParam);
+    hue_1RGB_hip_compute(&(pix_f24->f1[ 1]), &(pix_f24->f1[ 9]), &(pix_f24->f1[17]), hueParam);
+    hue_1RGB_hip_compute(&(pix_f24->f1[ 2]), &(pix_f24->f1[10]), &(pix_f24->f1[18]), hueParam);
+    hue_1RGB_hip_compute(&(pix_f24->f1[ 3]), &(pix_f24->f1[11]), &(pix_f24->f1[19]), hueParam);
+    hue_1RGB_hip_compute(&(pix_f24->f1[ 4]), &(pix_f24->f1[12]), &(pix_f24->f1[20]), hueParam);
+    hue_1RGB_hip_compute(&(pix_f24->f1[ 5]), &(pix_f24->f1[13]), &(pix_f24->f1[21]), hueParam);
+    hue_1RGB_hip_compute(&(pix_f24->f1[ 6]), &(pix_f24->f1[14]), &(pix_f24->f1[22]), hueParam);
+    hue_1RGB_hip_compute(&(pix_f24->f1[ 7]), &(pix_f24->f1[15]), &(pix_f24->f1[23]), hueParam);
+}
+
+__device__ void hue_hip_compute(uchar *srcPtr, d_float24 *pix_f24, float *hueParam)
+{
+    hue_8RGB_hip_compute(pix_f24, hueParam);
+}
+__device__ void hue_hip_compute(float *srcPtr, d_float24 *pix_f24, float *hueParam)
+{
+    hue_8RGB_hip_compute(pix_f24, hueParam);
+}
+__device__ void hue_hip_compute(half *srcPtr, d_float24 *pix_f24, float *hueParam)
+{
+    hue_8RGB_hip_compute(pix_f24, hueParam);
+}
+__device__ void hue_hip_compute(schar *srcPtr, d_float24 *pix_f24, float *hueParam)
+{
+    hue_8RGB_hip_compute(pix_f24, hueParam);
+}
+
+template <typename T>
+__global__ void hue_pkd_hip_tensor(T *srcPtr,
+                                   uint2 srcStridesNH,
+                                   T *dstPtr,
+                                   uint2 dstStridesNH,
+                                   float *hueTensor,
+                                   RpptROIPtr roiTensorPtrSrc)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if ((id_y >= roiTensorPtrSrc[id_z].xywhROI.roiHeight) || (id_x >= roiTensorPtrSrc[id_z].xywhROI.roiWidth))
+        return;
+
+    uint srcIdx = (id_z * srcStridesNH.x) + ((id_y + roiTensorPtrSrc[id_z].xywhROI.xy.y) * srcStridesNH.y) + ((id_x + roiTensorPtrSrc[id_z].xywhROI.xy.x) * 3);
+    uint dstIdx = (id_z * dstStridesNH.x) + (id_y * dstStridesNH.y) + id_x * 3;
+
+    float hueParam = (((int)hueTensor[id_z]) % 360) * (6.0f/360.0f);
+
+    d_float24 pix_f24;
+
+    rpp_hip_load24_pkd3_and_unpack_to_float24_pln3(srcPtr + srcIdx, &pix_f24);
+    hue_hip_compute(srcPtr, &pix_f24, &hueParam);
```

Comment:
Please directly call hue_8rgb_hip_compute() here. Since templated U8/F16/F32/I8 need not have different ops, we really don't need the 4 versions from L58-L73 above.
Also, no need to pass srcPtr. Just &pix_f24, &hueParam should suffice